pkg/controllers/managementuser/healthsyncer/healthsyncer.go

```diff
@@ -85,6 +95,19 @@ func (h *HealthSyncer) getComponentStatus(cluster *v3.Cluster) error {
 	sort.Slice(cluster.Status.ComponentStatuses, func(i, j int) bool {
 		return cluster.Status.ComponentStatuses[i].Name < cluster.Status.ComponentStatuses[j].Name
 	})
+
+	if excludedComponents, ok := excludedComponentMap[cluster.Status.Driver]; ok {
+		for i := 0; i < len(excludedComponents); i++ {
+			for ii := 0; ii < len(cluster.Status.ComponentStatuses); ii++ {
+				if cluster.Status.ComponentStatuses[ii].Name == excludedComponents[i] {
+					cluster.Status.ComponentStatuses = append(cluster.Status.ComponentStatuses[:ii], cluster.Status.ComponentStatuses[ii+1:]...)
+					break
+				}
+			}
+
+		}
+	}
```

Comment:
This could be dropped and add the below at line 91, this should save us some processing/loops
```
for _, cs := range cses.Items {
		if excludedComponents, ok := excludedComponentMap[cluster.Status.Driver]; ok && !slice.ContainsString(excludedComponents, cs.Name) {
			clusterCS := convertToClusterComponentStatus(&cs)
			cluster.Status.ComponentStatuses = append(cluster.Status.ComponentStatuses, *clusterCS)
		}
	}
```